cdap-app-templates/cdap-etl/cdap-etl-proto/src/main/java/io/cdap/cdap/etl/proto/v2/spec/StageSpec.java

```diff
@@ -315,4 +318,18 @@ public int hashCode() {
       return Objects.hash(port, schema);
     }
   }
+
+  /**
+   * Returns a copy of the StageSpec object with {@link StageSpec#maxPreviewRecords} field updated
+   */
+  public static StageSpec createCopyWithUpdatedMaxPreviewRecords(StageSpec stageSpec, int newMaxPreviewRecords) {
+    if (newMaxPreviewRecords < stageSpec.getMaxPreviewRecords()) {
+      LOG.warn("Max preview records exceeds allowed limit. Setting maximum preview records to %s ",
+               newMaxPreviewRecords);
```

Comment:
LOG the specified preview records from stage spec as well.

Reply:
Use `{}` as placeholder, not `%s`